lib/phoenix/test/channel_test.ex

```diff
@@ -291,21 +307,21 @@ defmodule Phoenix.ChannelTest do
   Useful for testing UserSocket authentication. Returns
   the result of the handler's `connect/3` callback.
   """
-  defmacro connect(handler, params, connect_info \\ quote(do: %{})) do
+  defmacro connect(handler, params, connect_info \\ quote(do: %{}), options \\ []) do
```

Comment:
I find two optional arguments a smell, so what if we do this:

```
  defmacro connect(handler, params, options \\ quote(do: [])) do
```

And then `__connect__` we do:

```
  {connect_info, options} =
    if is_map(options) do
      IO.warn("Passing \"connect_info\" directly to connect/3 is deprecated, please pass \"connect_info: ...\" as an option instead")
      {options, []}
    else
      Keyword.pop(options, :connect_info, %{})
    end
```